app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/MessageFragment.java

```diff
@@ -1,64 +1,139 @@
-        contact.token = "12";
-        contacts.add(contact);
+    /**
+     * Set all listeners from MessageFragment
+     */
+    private void setListeners() {
+        binding.addContactBtn.setOnClickListener(v -> {
+            startActivity(new Intent(getContext(), AddContactActivity.class));
+        });
+    }
+
+    /**
+     * Get all contacts from user and display recyclerView
+     */
+    private void getContacts() {
+        loading(true);
+        // TODO: use cached contacts list once we have it
+        // Prevent contact display offline
+        User currUser = ((BaseApp) getActivity().getApplication()).getCurrUser();
+        if (currUser == null) {
+            loading(false);
+            displayError();
+            return;
+        }
+
+        ((MainActivity) getActivity()).getIdlingResource().increment();
+        String userUid = currUser.getUid();
+        db.collection(Constants.KEY_COLLECTION_USERS)
+                .document(userUid)
+                .collection(Constants.KEY_COLLECTION_CONTACTS)
+                .get()
+                .addOnCompleteListener(task -> {
+                    loading(false);
+                    if (task.isSuccessful() && task.getResult() != null) {
+                        contacts = new ArrayList<>();
+                        for (QueryDocumentSnapshot queryDocumentSnapshot: task.getResult()) {
+                            Contact contact = new Contact();
+                            contact.username = queryDocumentSnapshot.getString(Constants.KEY_USERNAME);
+                            contact.email = queryDocumentSnapshot.getString(Constants.KEY_EMAIL);
+                            contact.image = queryDocumentSnapshot.getString(Constants.KEY_IMAGE);
+                            contacts.add(contact);
```

Comment:
Maybe refactoring a bit of those kind of firebase requests function in some smaller ones would make it clearer